pkgs/tools/security/rng-tools/default.nix

```
@@ -1,24 +1,47 @@
-{ stdenv, fetchurl }:
-
+{ stdenv, fetchFromGitHub, libtool, autoconf, automake, pkgconfig
+, libsysfs
+  # WARNING: DO NOT USE BEACON GENERATED VALUES AS SECRET CRYPTOGRAPHIC KEYS
+  # https://www.nist.gov/programs-projects/nist-randomness-beacon
+, curl ? null, libxml2 ? null, openssl ? null, withNistBeacon ? false
+  # Systems that support RDRAND but not AES-NI require libgcrypt to use RDRAND as an entropy source
+, libgcrypt ? null, withGcrypt ? true
+, jitterentropy ? null, withJitterEntropy ? true
+}:
+with stdenv.lib;
 stdenv.mkDerivation rec {
-  name = "rng-tools-5";
+  name = "rng-tools-${version}";
+  version = "6.6";
 
-  src = fetchurl {
-    url = "mirror://sourceforge/gkernel/${name}.tar.gz";
+  src = fetchFromGitHub {
+    owner = "nhorman";
+    repo = "rng-tools";
 
-    sha256 = "13h7lc8wl9khhvkr0i3bl5j9bapf8anhqis1lcnwxg1vc2v058b0";
+    rev = "v${version}";
+    sha256 = "0c32sxfvngdjzfmxn5ngc5yxwi8ij3yl216nhzyz9r31qi3m14v7";
   };
 
+  nativeBuildInputs = [ libtool autoconf automake pkgconfig ];
+
+  preConfigure = "./autogen.sh";
+
+  configureFlags =
+       optional (!withJitterEntropy) "--disable-jitterentropy"
+    ++ optional (!withNistBeacon) "--without-nistbeacon"
+    ++ optional (!withGcrypt) "--without-libgcrypt";
+
+  buildInputs = [ libsysfs ]
+    ++ optional withJitterEntropy [ jitterentropy ]
+    ++ optional withGcrypt [ libgcrypt ]
```

```suggestion
    ++ optional withGcrypt [ libgcrypt.dev ]
```

Otherwise the headers are not found during configurePhase